projects/components/src/page-time-range/page-time-range.component.ts

```diff
@@ -19,13 +19,15 @@ export class PageTimeRangeComponent {
     const activatedRoute = this.navigationService.getCurrentActivatedRoute();
     const urlSegments: UrlSegment[] = activatedRoute.pathFromRoot.flatMap(activeRoute => activeRoute.snapshot.url);
 
-    if (this.shouldSavePageTimeRange(activatedRoute)) {
+    if (this.shouldSavePageTimeRange()) {
       this.savePageTimeRange(selectedTimeRange, urlSegments[0]);
     }
   }
 
-  public shouldSavePageTimeRange(route: ActivatedRoute): boolean {
-    return !isNil(route.snapshot.data?.defaultTimeRange);
+  public shouldSavePageTimeRange(): boolean {
```

Comment:
nit: want to take in an activated route rather than have it resolved inside - always strive to make functions pure - they're easier to understand and easier to test.